Build helm values schemas with kubernetes 1.29 resources

mathiaHT opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->

closes: #38453

Build helm chart/values.schemas.json definitions using kubernetes 1.29. 

The initial objective was to use the kubernetes sidecar feature: https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/

---

Commit:
update doc

Commit:
exlude chart schemas

Commit:
autoscaling.v2beta2 resources are now v2

Commit:
fix helm tests for restartPolicy

potiuk reviewed (approved):
LGTM. @jedcunningham ? @raphaelauv ?

jscheffl committed:
Allow users to write dag_id and task_id in their national characters, added display name for dag / task (v2) (#38446)

* Add display name in DAGs and tasks

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Aleksandr Shelukheev <shelukheev@gmail.com>

* Fix frontend issues

* Fix pytests

* Add DAG display tool tips

* Implement national language char display example oon two DAGs

* Review nit from other PR

* Fix WWW pytest on HTML code

* Review feedback, make task_display_name a property, not a field in mapped operator

* Review feedback, rename internal fields

* Small nit, optimize sorting on DAG home page with display name

* Review feedback

* Fix pytests in API plus review feedback on API

* Review feedback, extend schema, update apispecs and fix breadcrumb navigation

* nit, update page titles as well

* Fix pytests for extended API model

* Ensure mapped operator also provides a display string

* Ensure display field is only serialized if different from task id

* fix pytest for API

* Review feedback from TP

* Move labels down to base and mapped operator

* Move labels up to abstract operator

* Only one decorator is needed

---------

Co-authored-by: Vincent Gao <xgao1023@gmail.com>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Aleksandr Shelukheev <shelukheev@gmail.com>

hussein-awala committed:
Add a migration script for encrypted trigger kwargs (#38358)

ferruzzi committed:
Add executor field to the DB and parameter to the operators (#38474)

This field will be used by AIP-61 code to store and retrieve which
executor a task is meant to run on.

---------

Co-authored-by: Niko Oliveira <onikolas@amazon.com>

hussein-awala reviewed (approved):
Most of the updated resources were deprecated in 1.26+.

In general, I wonder if we should use the max-supported version or the min-supported version to generate the schema. For this PR, it looks like all updated versions are already present in 1.26, but not sure if it will be always the case.

LGTM

Commit:
update doc assets

mathiaHT committed:
Merge branch 'apache:main' into main
